core/trino-main/src/main/java/io/trino/operator/ChannelSet.java

```diff
@@ -53,67 +53,69 @@ public boolean isEmpty()
 
     public boolean containsNull()
     {
-        return containsNull;
+        return set.containsNull();
     }
 
-    public boolean contains(int position, Page page)
+    public boolean contains(Block valueBlock, int position)
     {
-        return hash.contains(position, page);
+        return set.contains(valueBlock, position);
     }
 
-    public boolean contains(int position, Page page, long rawHash)
+    public boolean contains(Block valueBlock, int position, long rawHash)
     {
-        return hash.contains(position, page, rawHash);
+        return set.contains(valueBlock, position, rawHash);
     }
 
     public static class ChannelSetBuilder
     {
-        private final Type type;
-        private final OperatorContext operatorContext;
         private final LocalMemoryContext localMemoryContext;
-        private final GroupByHash hash;
+        private final FlatSet set;
```

Comment:
With this change you no longer allow for a fallback switch to old implementation based on `MultiChannelGroupByHash` via `legacy.flat-group-by-hash`. 
Safer approach would be to have two implementations of `ChannelSet` - not sure if really worth the fuss - depends on how confident you are this will not break :)

Reply:
The fallback is going to be removed next week anyway, so I'm not sure it matters.